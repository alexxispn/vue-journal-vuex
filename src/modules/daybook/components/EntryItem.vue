<template>
  <div class="entry-container mb-3 pointer p-2" @click="goToEntry">
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">{{day}}</span>
      <span class="mx-1 fs-5">{{month}}</span>
      <span class="mx-2 fw-light">{{year}}, {{weekday}}</span>
    </div>

    <div class="entry-description">
      <p class="text-muted">
        {{ shortText }}
      </p>
    </div>
  </div>
</template>

<script>
import getFullDate from "@/modules/daybook/helpers/getFullDate";

export default {
  name: "EntryItem",
  props: {
    entry: {
      type: Object,
      required: true
    }
  },
  computed: {
    shortText() {
      return (this.entry.text.length > 130)
          ? this.entry.text.substr(0, 130) + '...'
          : this.entry.text;
    },
    day() {
      const {day} = getFullDate(this.entry.date);
      return day;
    },
    month() {
      const {month} = getFullDate(this.entry.date);
      return month;
    },
    year() {
      const {year} = getFullDate(this.entry.date);
      return year;
    },
    weekday() {
      const {weekday} = getFullDate(this.entry.date);
      return weekday;
    }
  },
  methods: {
    goToEntry() {
      this.$router.push({
        name: 'db-entry',
        params: {
          id: this.entry.id
        }
      });
    }
  }
}
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: all 0.3s ease-in-out;

  &:hover {
    background-color: lighten(grey, 45%);
    transition: all 0.3s ease-in-out;
  }

  .entry-title {
    .text-success {
      font-size: 0.8rem;
    }
  }
}
</style>
