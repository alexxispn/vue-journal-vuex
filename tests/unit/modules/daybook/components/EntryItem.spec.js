import EntryItem from "@/modules/daybook/components/EntryItem";
import {shallowMount} from "@vue/test-utils";
import {journalState} from "../../../mock-data/journal-state";

describe("EntryItem component", () => {
    const mockRouter = {
        push: jest.fn()
    }
    const wrapper = shallowMount(EntryItem, {
        props: {
            entry: journalState.entries[1]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    });

    it("should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
    it("should redirect to entryView page when entry is clicked", () => {
        wrapper.find(".entry-container").trigger("click");
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: "entry",
            params: {
                id: journalState.entries[1].id
            }
        });
    });
    it ("should show the correct date", () => {
        expect(wrapper.vm.day).toBe(4);
        expect(wrapper.vm.month).toBe("October");
        expect(wrapper.vm.year).toBe(2022);
        expect(wrapper.vm.weekday).toBe("Tuesday");
    });
});
