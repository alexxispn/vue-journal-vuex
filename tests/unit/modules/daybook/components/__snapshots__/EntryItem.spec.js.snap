// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryItem component should match snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">4</span><span class="mx-1 fs-5">October</span><span class="mx-2 fw-light">2022, Tuesday</span></div>
  <div class="entry-description">
    <p class="text-muted">Hola esto es una prueba con las envs</p>
  </div>
</div>
`;
